src/cinematic.cs

```diff
@@ -0,0 +1,494 @@
+		NSNumber [] VideoCompositionTrackIds { get; }
+
+		[Export ("sampleDataTrackIDs", ArgumentSemantic.Strong)]
+		NSNumber [] SampleDataTrackIds { get; }
+	}
+
+	[TV (17, 0), NoWatch, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (CNAssetInfo))]
+	interface CNCompositionInfo {
+		[Export ("insertTimeRange:ofCinematicAssetInfo:atTime:error:")]
+		bool InsertTimeRange (CMTimeRange timeRange, CNAssetInfo assetInfo, CMTime startTime, [NullAllowed] out NSError outError);
+	}
+
+	[TV (17, 0), NoWatch, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CNRenderingSessionAttributes {
+		[Async]
+		[Static]
+		[Export ("loadFromAsset:completionHandler:")]
+		void Load (AVAsset asset, Action<CNRenderingSessionAttributes, NSError> completionHandler);
+
+		[Export ("renderingVersion")]
+		nint RenderingVersion { get; }
+	}
+
+	[TV (17, 0), NoWatch, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CNRenderingSessionFrameAttributes : NSCopying, NSMutableCopying {
+		[Export ("initWithSampleBuffer:sessionAttributes:")]
+		NativeHandle Constructor (CMSampleBuffer sampleBuffer, CNRenderingSessionAttributes sessionAttributes);
+
+		[Export ("initWithTimedMetadataGroup:sessionAttributes:")]
+		NativeHandle Constructor (AVTimedMetadataGroup metadataGroup, CNRenderingSessionAttributes sessionAttributes);
+
+		[Export ("focusDisparity")]
+		float FocusDisparity { get; set; }
+
+		[Export ("fNumber")]
+		float FNumber { get; set; }
+	}
+
+	[TV (17, 0), NoWatch, Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CNRenderingSession {
+		[Export ("initWithCommandQueue:sessionAttributes:preferredTransform:quality:")]
+		NativeHandle Constructor (IMTLCommandQueue commandQueue, CNRenderingSessionAttributes sessionAttributes, CGAffineTransform preferredTransform, CNRenderingQuality quality);
+
+		[Export ("commandQueue", ArgumentSemantic.Strong)]
+		IMTLCommandQueue CommandQueue { get; }
+
+		[Export ("sessionAttributes", ArgumentSemantic.Strong)]
+		CNRenderingSessionAttributes SessionAttributes { get; }
+
+		[Export ("preferredTransform")]
+		CGAffineTransform PreferredTransform { get; }
+
+		[Export ("quality")]
+		CNRenderingQuality Quality { get; }
+
+		[Export ("encodeRenderToCommandBuffer:frameAttributes:sourceImage:sourceDisparity:destinationImage:")]
+		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, CVPixelBuffer destinationImage);
+
+		[Export ("encodeRenderToCommandBuffer:frameAttributes:sourceImage:sourceDisparity:destinationRGBA:")]
+		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, IMTLTexture destinationRGBA);
```

Comment:
```suggestion
		bool EncodeRender (IMTLCommandBuffer commandBuffer, CNRenderingSessionFrameAttributes frameAttributes, CVPixelBuffer sourceImage, CVPixelBuffer sourceDisparity, IMTLTexture destinationRgba);
```
not quite sure if I remember the rule, but should this be as such?